Add o3de-extras automatic test; WIP manual tests (#13)
RobotecAI/o3de-ros2-gem-testing: o3de-extras-repo/Checklist.md

- [ ] `Ros2ProjectTemplate` template correctly generates the project
- [ ] `Ros2RoboticManipulationTemplate` template correctly generates the project
- [ ] `ProteusRobot` Assets Gem: all assets are build
- [ ] `ROS2` Gem: builds on Ubuntu 22.04 with ROS 2 Humble in profile
- [ ] `ROS2` Gem: builds on Ubuntu 22.04 with ROS 2 Humble in monolithic release
- [ ] `ROS2` Gem: builds on Ubuntu 24.04 with ROS 2 Jazzy in profile
- [ ] `RosRobotSample` Assets Gem: all assets are build
- [ ] `WarehouseAssets` Assets Gem: all assets are build
- [ ] `WarehouseAutomation` Gem: builds on Ubuntu 24.04 with ROS 2 Jazzy in profile
- [ ] `WarehouseSample` Assets Gem: all assets are build
- [ ] ROS 2 workspace of the `Ros2FleetRobotTemplate` based project builds correctly
- [ ] unit tests in `ROS2` Gem completed on Ubuntu 22.04 with ROS 2 Humble
- [ ] unit tests in `ROS2` Gem completed on Ubuntu 24.04 with ROS 2 Jazzy

## Manual tests

My test setup:
- Ubuntu version: *e.g. 22.04*
- ROS 2 version: *e.g. Humble*
- compiler version: *e.g. clang 18.1.3*
- O3DE version: *e.g. 2409.2_nightly_build from January 14th*
- used SDK build: *yes*

### Ros2FleetRobotTemplate tests

> - [ ] 1A. I checked if any warnings are presented when starting the O3DE Editor

Author: I primarily meant the warnings displayed as a pop-up window when starting the Editor. Such warnings have typically the biggest impact on the projects (e.g. missing components/meshes/prefabs).

> - [ ] 1A. I checked if any warnings are presented when starting the O3DE Editor

Reviewer: ```suggestion
- [ ] 1A. I checked if any errors are presented when starting the O3DE Editor
```

> - [ ] 1A. I checked if any warnings are presented when starting the O3DE Editor

Reviewer 2: Technically, these warnings pop out when loading the level, not the editor itself. The Editor might load just fine, but there will be warnings in levels.

(also applies to other templates)

```suggestion
- [ ] 1A. I checked if any warnings are presented when loading a level in the O3DE Editor
```

> - [ ] 1A. I checked if any warnings are presented when starting the O3DE Editor

Reviewer 2: We should check both pop-up warnings (levels) and editor starting errors (like missing libs).

I think it is technically possible to miss the library and don't notice it.

- [ ] 1A. I checked if there are any missing meshes and textures
- [ ] 1B. I checked if `o3de_fleet_nav_launch.py` launcher works correctly
- [ ] 1B. I saw Lidar data in RViz2 windows when launching `o3de_fleet_nav_launch.py` launcher
- [ ] 1B. I checked if all expected ROS 2 topics are present
- [ ] 1B. I checked if spawnable prefabs and spawnpoints are presented correctly via ROS 2

### Ros2ProjectTemplate tests

> - [ ] 2A. I checked if any warnings are presented when starting the O3DE Editor

Reviewer: ```suggestion
- [ ] 2A. I checked if any errors are presented when starting the O3DE Editor
```

- [ ] 2A. I checked if there are any missing meshes and textures
- [ ] 2B. I checked if `navigation.launch.py` launcher works correctly
- [ ] 2B. I saw Lidar data in RViz2 window when launching `navigation.launch.py` launcher

> - [ ] 2B. I checked if spawnable prefabs and spawnpoints are presented correctly via ROS 2

Reviewer 3: This position is not listen in Action/Outcome guidelines in manual tests description.

- [ ] 2B. I checked if all expected ROS 2 topics are present

### Ros2RoboticManipulationTemplate tests
- [ ] 3A. I checked if any warnings are presented when starting the O3DE Editor in two different levels
- [ ] 3A. I checked if there are any missing meshes and textures in two different levels
- [ ] 3B. I checked if articulations can be controlled using ROS 2 action
- [ ] 3B. I checked if `panda_moveit_config_demo.launch.py` launcher works correctly

> - [ ] 3B. I saw camera data in RViz2 windows when launching `panda_moveit_config_demo.launch.py` launcher

Reviewer 3: Maybe an additional line here, because there is a corresponding Action/Outcome in tests description?
```suggestion
- [ ] 3B. I saw camera data in RViz2 windows when launching `panda_moveit_config_demo.launch.py` launcher
- [ ] 3B. I checked if robotic arm can be controlled using RViz2 tool
```
Will be an explicit statement that the tester checked that robots at least respond to commands. If this will be fine for you, I would also consider similar checkboxes in Ros2FleetRobotTemplate and Ros2ProjectTemplate, where setting navigation goals is suggested.
